tools/c7n_azure/tests/test_sqldatabase.py

```diff
@@ -257,3 +264,242 @@ def test_filter_database_with_retention_period_unit_mismatch(self):
 
         resources = p.run()
         self.assertEqual(len(resources), 0)
+
+
+class ShortTermBackupRetentionPolicyActionSchemaTest(BaseTest):
+
+    def test_schema_with_valid_short_term_retention_period_values(self):
+        for period in ShortTermBackupRetentionPolicyAction.VALID_RETENTION_PERIOD_DAYS:
+            self._test_retention_period_days(days=period, valid=True)
+
+    def test_schema_with_invalid_short_term_retention_period_values(self):
+        for period in [5, 10, 20, 100]:
+            self._test_retention_period_days(days=period, valid=False)
+
+    def _test_retention_period_days(self, days, valid=True):
+
+        def do_validation():
+            with self.sign_out_patch():
+                p = self.load_policy({
+                    'name': 'short-term-action-schema-validate',
+                    'resource': 'azure.sqldatabase',
+                    'actions': [
+                        {
+                            'type': 'update-short-term-backup-retention-policy',
+                            'retention-period-days': days
+                        }
+                    ]
+                }, validate=True)
+                return p
+
+        if valid:
+            p = do_validation()
+            self.assertTrue(p, "Expected {} to be a valid retention period".format(days))
+        else:
+            with self.assertRaises(PolicyValidationError,
+                    msg="Expected {} to be an invalid retention period".format(days)):
+                do_validation()
+
+
+# NOTE: Normally, updating the retention policy on a DB requires ARM polling to know when the
+# operation has finished. However, this polling happens client-side and causes the tests to complete
+# slowly. In order to speed these up, the cassettes were manually modified to immediately return the
+# completed operation.
+class ShortTermBackupRetentionPolicyActionTest(BaseTest):
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super(ShortTermBackupRetentionPolicyActionTest, cls).setUpClass(*args, **kwargs)
+        cls.client = local_session(Session).client('azure.mgmt.sql.SqlManagementClient') \
+            .backup_short_term_retention_policies
+
+        cls.retention_policy_context = [
+            "test_sqlserver",
+            "cctestsqlserverideeowhs5iudo",
```

Comment:
suffix will be different for a different subscription

Reply:
ah, right